pkg/factory/risingwave_object_factory.go

```diff
@@ -1084,7 +1084,7 @@ func (f *RisingWaveObjectFactory) envsForS3() []corev1.EnvVar {
 		endpoint = strings.ReplaceAll(endpoint, "${REGION}", fmt.Sprintf("$(%s)", envs.S3CompatibleRegion))
 		endpoint = strings.ReplaceAll(endpoint, "${BUCKET}", fmt.Sprintf("$(%s)", envs.S3CompatibleBucket))
 
-		if !strings.HasPrefix(endpoint, "https://") {
+		if !strings.HasPrefix(endpoint, "https://") && !strings.HasPrefix(endpoint, "http://") {
```

Comment:
Good catch! Could you help add a test case right under the [`"s3-compatible"`](https://github.com/risingwavelabs/risingwave-operator/blob/86a79f61fb2c1be7e7ae6a73236281829f048b4a/pkg/factory/risingwave_object_factory_testcases_test.go#L3165)?